synapse/handlers/auth.py

```diff
@@ -708,6 +715,7 @@ def _check_local_password(self, user_id, password):
 
         Args:
             user_id (str): complete @user:id
+            password (unicode): the provided password
```

Comment:
I think it might be worth clarifying that it will be a `str` on python3. Something like:

```
password (str|unicode): the provided password. On python2, *must* be a unicode.
```

(and similar on `validate_hash` etc etc)

Of course what's really happening is that elsewhere we have been sloppy about saying `str` when we mean `str|unicode`, though I don't suggest we change that...

Reply:
It's clearer to use `bytes` (flat bytes), `str` (str on either platform), or `unicode` (unicode), even though there is no such thing as `unicode` on Python 3.

Reply:
ok, as discussed, `unicode` seems the right answer here. However, the difference with `user_id` and the return type is now very striking; suggest you update them too.